### PR TITLE
Malickc/gwm update

### PR DESCRIPTION
**Pull request by codymalick**
Rake is passing, updates to ChefSpec for code quality

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -29,7 +29,8 @@
 end
 
 no_clone = node.chef_environment == 'vagrant' &&
-           ::File.directory?(::File.join(node['ganeti_webmgr']['path'], '.git'))
+           ::File.directory?(::File.join(node['ganeti_webmgr']['path'],
+                                         '.git'))
```
Comment (ramereth) [inline recipes/default.rb]: You don't seem to be using our updated rubocop.yml config. Please update it and this change shouldn't be needed.

**Comment by ramereth** [comment]
@codymalick are you still working on this? I can't get the default recipe to config on CentOS 6.

**Comment by ramereth** [comment]
So I did some investigative work on this cookbook on the master branch. The only changes I needed to make to get it to work was the following (Note: for the mysql suite):

metadata.rb:
``` diff
diff --git a/metadata.rb b/metadata.rb
index b87ddd6..658be76 100644
--- a/metadata.rb
+++ b/metadata.rb
@@ -11,11 +11,11 @@ depends          'build-essential'
 depends          'python'
 depends          'git'
 depends          'nginx'
-depends          'mysql'
+depends          'mysql', '~> 3.0.12'
 depends          'openssl'
 depends          'postgresql'
 depends          'sqlite'
-depends          'database'
+depends          'database', '~> 1.4.0'
 depends          'hostsfile'
 depends          'apache2'
 depends          'runit'
```

Basically, it's relying on some really old versions of upstream cookbooks. This goes against what we currently have locked in production so we have two options:

1. Create a new gwm environment with these two locks (yuck)
2. Apply fixes to the recipes that work with the versions of these cookbooks we have locked in production currently (database-4.0.6 / mysql-5.3.6).

I'd probably prefer going to 2 for better long term support but option 1 is probably the path of least resistance in the short term. I am not sure how much work is involved for 2 but I don't *think* that much.

In addition, I manually installed Twisted-15.4.0 which matches what's running in production. The version that gets pulled in now doesn't support Python 2.6 but 15.4.0 does. I've created osuosl/ganeti_webmgr#103 to address that problem moving forward as it should have been locked in the first place. Once that gets merged, then that should resolve the Python 2.6 issues.

Some other things I've remembered: I don't think the default suite has ever worked however the mysql suite did. I'd recommend you just replace the current default suite with the mysql suite and use that as the default suite moving forward. That would totally scrap trying to get sqlite working. 

I would scrap the SCL approach entirely as well since that should be fixed with the older version of Twisted.

**Comment by ramereth** [comment]
@codymalick can we go ahead and close this in favor of what we did before?

**Comment by jldugger** [comment]
@codymalick @ramereth I'm closing this PR, since really, we need to redo enough stuff in this cookbook that conflicts with any fixups extracted would be difficult.

**Pull request closed without merging**